### ISSUE TITLE
Some points are closer than expected

### ISSUE THREAD

**Issue by arcanis** [Kromey/fast_poisson#33]
```
fn main() {
    let seed: u64 = 6980462275800279379;

    let points = fast_poisson::Poisson2D::new()
        .with_dimensions([30.0, 20.0], 5.0)
        .with_seed(seed)
        .generate();

    println!("{:?}", points);
}
```

The code above will generate the following points:

```
[[8.646205749465315, 4.564256909835271], [13.372778272226896, 1.5794059185135163], [18.602230865277487, 1.1982050375675521], [13.05736613529966, 7.221287582463809], [20.48940118239967, 6.569870681868846], [3.811709281568241, 1.3564982684349047], [4.075533167429171, 8.685442353142616], [0.18763045829584257, 12.357980978471723], [22.773615127446988, 11.96743956447334], [29.989267617289595, 15.011834274792559], [10.67655345880279, 15.697672434048096], [17.377151009951923, 13.878320967917393], [4.888297043691209, 14.259382372300996], [4.542168437002315, 19.744919432349963], [22.259603340322233, 19.8389668782879], [27.426705719963394, 0.8054627187756263], [15.676328098136594, 18.743276062899845], [24.971360078286544, 17.265061003214317], [19.334384056771537, 1.2177481591250694], [19.236686200135065, 19.303667361679185], [27.04042942369734, 7.2962803329058055]]
```

The two interesting points are those:

```
[18.602230865277487, 1.1982050375675521]
[19.334384056771537, 1.2177481591250694]
```

Their distance from each other is 0.7324139740714496, far less than the expected threshold of 5.0 - is that expected? This happens relatively often, about once every ten different seeds for me.

**Comment by Kromey** [comment]
Thanks for your code to reproduce the issue, I've used it to create a new test that confirms the bug - you're exactly right that those two points should not exist so close to each other. I'm beginning to dig into the code now to see if I can figure out what the issue is.

**Comment by Kromey** [comment]
It would seem that the cause is an issue with finding neighboring "cells" in the spatial grid I implemented. Currently testing a possible fix utilizing a k-d tree, however in doing so I accidentally uncovered #34, which I'm fairly certain is not caused by the use of the k-d tree and was merely coincidentally found while more extensively testing seeds to ensure a fix for this issue.

**Comment by Kromey** [comment]
@arcanis if you want to use the current `main` branch now your issue is resolved. I expect to have a fix for #34 in time to publish a new release tomorrow, possibly even later today if I get really lucky :four_leaf_clover:

**Comment by Kromey** [comment]
FYI @arcanis I've just published version 1.0.0 to crates.io, with this and a few other fixes and general improvements
